mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -697,3 +698,74 @@ def __repr__(self):
         repr_str += '(bbox_enlarge_range={})'.format(
             self.bbox_enlarge_range.tolist())
         return repr_str
+
+
+@PIPELINES.register_module()
+class VoxelBasedPointSampler(object):
+    """Voxel based point sampler.
+
+    Apply voxel sampling to multiple sweep points.
+
+    Args:
+        name (str): Name of the dataset.
+        num_points (int): Number of points to be sampled.
+    """
+
+    def __init__(self, cur_sweep_cfg, prev_sweep_cfg=None):
+        self.cur_voxel_num = cur_sweep_cfg.pop('random_voxel_num')
+        self.cur_sweep_sampler = VoxelGenerator(**cur_sweep_cfg)
+        if prev_sweep_cfg is not None:
+            self.prev_voxel_num = prev_sweep_cfg.pop('random_voxel_num')
+            self.prev_sweep_sampler = VoxelGenerator(**prev_sweep_cfg)
+        else:
+            self.prev_voxel_num = 0
+
+    def __call__(self, results):
+        """Call function to sample points from multiple sweeps.
+
+        Args:
+            input_dict (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Results after sampling, 'points', 'pts_instance_mask' \
+                and 'pts_semantic_mask' keys are updated in the result dict.
+        """
+        points = results['points']
+        cur_points_num = results['cur_points_num']
```

Comment:
Do not use `cur_points_num`, use time encodings to find the points since the points may be shuffled.